pallene/to_ir.lua

```diff
@@ -520,6 +486,27 @@ function ToIR:convert_stat(cmds, stat)
     elseif tag == "ast.Stat.Break" then
         table.insert(cmds, ir.Cmd.Break())
 
+    elseif tag == "ast.Stat.Func" then
+        local prev_f_id = 1
```

Comment:
Maybe we should refactor the `enter_func` to allow entering another function while inside another function. When `enter_func` is called, take the current values of `self.func`, etc, that refer to the outer function and save them  to a stack. Then restore those variables when the inner function exits.

I really don't like that currently we have a `self:enter_function(prev_f_id)` without a matching `self:exit_function`.

Reply:
Yes, I thought about that, but didn't want to perform yet another "big change", this PR seems big enough as is. However, I totally agree with you, this is a ghastly workaround.
Maybe I should refactor it on a separate commit.

Reply:
If we aren't refactoring that on this PR, could you please create a separate issue for it, so that we don't forget?

Reply:
@hugomg https://github.com/pallene-lang/pallene/issues/363